packages/salesforce-adapter/src/filters/workflow_rule.ts

```diff
@@ -0,0 +1,161 @@
+/*
+*                      Copyright 2020 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import _ from 'lodash'
+import { collections } from '@salto-io/lowerdash'
+import { logger } from '@salto-io/logging'
+import {
+  Element, ReferenceExpression, ElemID, Value, Values, isReferenceExpression,
+} from '@salto-io/adapter-api'
+import {
+  findInstances,
+} from '@salto-io/adapter-utils'
+import { FilterCreator } from '../filter'
+import { parentApiName } from './utils'
+import { groupByAPIName, ApiNameMapping } from './instance_references'
+import {
+  SALESFORCE, API_NAME_SEPERATOR,
+  WORKFLOW_ACTION_ALERT_METADATA_TYPE,
+  WORKFLOW_FIELD_UPDATE_METADATA_TYPE,
+  WORKFLOW_FLOW_ACTION_METADATA_TYPE,
+  WORKFLOW_OUTBOUND_MESSAGE_METADATA_TYPE,
+  WORKFLOW_RULE_METADATA_TYPE,
+  WORKFLOW_TASK_METADATA_TYPE,
+} from '../constants'
+
+const log = logger(module)
+const { makeArray } = collections.array
+
+export const WORKFLOW_RULE_TYPE_ID = new ElemID(
+  SALESFORCE,
+  WORKFLOW_RULE_METADATA_TYPE,
+)
+
+export type WorkflowActionType = 'Alert' | 'FieldUpdate' | 'FlowAction' | 'OutboundMessage' | 'Task'
+export const WORKFLOW_ACTION_TYPE_MAPPING: Record<WorkflowActionType, string> = {
+  Alert: WORKFLOW_ACTION_ALERT_METADATA_TYPE,
+  FieldUpdate: WORKFLOW_FIELD_UPDATE_METADATA_TYPE,
+  FlowAction: WORKFLOW_FLOW_ACTION_METADATA_TYPE,
+  OutboundMessage: WORKFLOW_OUTBOUND_MESSAGE_METADATA_TYPE,
+  Task: WORKFLOW_TASK_METADATA_TYPE,
+}
+
+
+export type WorkflowActionReference = {
+  name: string | ReferenceExpression
+  type: WorkflowActionType
+}
+
+type WorkflowRule = Values & {
+  actions: WorkflowActionReference | WorkflowActionReference[]
+}
+
+/* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+function isWorkflowActionReference(value: any): value is WorkflowActionReference {
+  return value !== undefined && (
+    (_.isString(value.name) || isReferenceExpression(value.name))
+    && Object.keys(WORKFLOW_ACTION_TYPE_MAPPING).includes(value.type)
+  )
+}
+
+/* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+function isWOrkflowRule(value: any): value is WorkflowRule {
+  return value !== undefined && (
+    (_.isArray(value.actions) && value.actions.every(isWorkflowActionReference))
+    || isWorkflowActionReference(value.actions)
+  )
+}
```

Comment:
similar comment to the one above about type guards
```suggestion
const isWorkflowActionReference(value: Values): value is WorkflowActionReference => (
  value !== undefined
  && (_.isString(value.name) || isReferenceExpression(value.name))
  && Object.keys(WORKFLOW_ACTION_TYPE_MAPPING).includes(value.type)
)

const isWorkflowRule(value: any): value is WorkflowRule => (
  value !== undefined && (
    (Array.isArray(value.actions) && value.actions.every(isWorkflowActionReference))
    || isWorkflowActionReference(value.actions)
  )
)
```